src/UXClient/Components/GeoProcessGraphic/GeoProcessGraphic.ts

```diff
@@ -0,0 +1,179 @@
+    (<HTMLElement>this.component.node()).style.width = `${this.width}px`;
+    (<HTMLElement>this.component.node()).style.height = `${this.height}px`;
+    
+     this.map = new atlas.Map(<HTMLElement>this.component.node(), {
+      authOptions: {
+        authType: atlas.AuthenticationType.subscriptionKey,
+        subscriptionKey: this.azureMapsSubscriptionKey
+      }
+    });
+    this.map.events.add('ready', () => {
+      this.dataSource = new atlas.source.DataSource();
+      this.map.sources.add(this.dataSource);
+
+      for (let i = 0; i < this.tsqExpressions.length; i++){
+        let popup = new atlas.Popup({
+          content: `<div class = 'tsi-gpgPopUp id= tsi-popup${i}'></div>`,
+          pixelOffset: [0, -30]
+        });
+        let marker = new atlas.HtmlMarker({
+          htmlContent: `<div class = tsi-geoprocess-graphic> <img class='tsi-gpgcircleImage 
+          id= tsi-htmlMarker${i}' src= "` + this.tsqExpressions[i].image + '" /> </div>',
+          position: [0,0],
+          popup: popup
+        });
+        this.map.markers.add(marker);
+        this.map.popups.add(popup);
+        this.map.events.add('click', marker, () => {
+          marker.togglePopup();
+        });
+      }
+    });
+    this.map.setCamera({
+        center: this.center,
+        bearing: this.bearing,
+        pitch: this.pitch,
+        zoom: this.zoom,
+        maxZoom: this.maxZoom,
+        minZoom: this.minZoom,
+        type: "fly",
+        duration: this.duration
+    });
+    return Promise.resolve({
+        graphic: this.map,
+        width: this.width,
+        height: this.height
+      });
+  }
+  
+  protected getDataPoints(promise: Array<IGeoProcessGraphicLabelInfo>){
+    let dataPoints = promise.map((r): IGeoProcessGraphicLabelInfo => {
+        return this.parseTsqResponse(r);  
+      });
+      this.updateDataMarkers(dataPoints);
+  }
+
+  protected parseTsqResponse(response) {
+    let parsedResponse = {};
+    if (response && response.properties){
+      for (let i =0; i<response.properties.length; i++){
+          response.properties[i] && response.properties[i].name && response.properties[i].values ?
+            parsedResponse[response.properties[i].name] = response.properties[i].values[0]
+              : null;
+      }
+    }
+    return parsedResponse;
+  }
+
+  protected updateDataMarkers(dataPoints: Array<any>) {
+      for(let i = 0; i < dataPoints.length; i++) {
+        let lat = dataPoints[i][this.tsqExpressions[i].positionXVariableName];
+        let lon = dataPoints[i][this.tsqExpressions[i].positionYVariableName];
+        this.map.markers.getMarkers()[i].setOptions({
+            position: [lat, lon]
+        });
+        let dataPointArr = Object.entries(dataPoints[i]);
+        
+        this.map.popups.getPopups()[i].setOptions({
+            position: [lat, lon],
+            content: this.createTable(dataPointArr, i)
+        });
+        console.log(this.map.popups.getPopups()[i]);
```

Comment:
You have a `console.log()` here, may be intentional, but just a heads up!

Reply:
that was accident, i took it off